src/adapters/db.ts

```diff
@@ -193,20 +193,18 @@ export function createDBComponent(components: Pick<AppComponents, 'pg' | 'logs'>
     async executeTx<T>(cb: (client: PoolClient) => Promise<T>): Promise<T> {
       const pool = pg.getPool()
       const client = await pool.connect()
-      let res: T | undefined
       await client.query('BEGIN')
       try {
-        const callbackResult = await cb(client)
+        const res = await cb(client)
         await client.query('COMMIT')
-        res = callbackResult
+        return res
       } catch (error) {
         logger.error(error as any)
         await client.query('ROLLBACK')
         client.release()
```

Comment:
not sure if this a problem, but you are releasing it twice, once here and one one finally